File: app/src/main/java/com/happs/ximand/ringcontrol/model/dao/AuthUtils.java
package com.happs.ximand.ringcontrol.model.dao;

import java.util.UUID;

import static android.os.Build.BOARD;
import static android.os.Build.BRAND;
import static android.os.Build.DEVICE;
import static android.os.Build.DISPLAY;
import static android.os.Build.HOST;
import static android.os.Build.ID;
import static android.os.Build.MANUFACTURER;
import static android.os.Build.MODEL;
import static android.os.Build.PRODUCT;
import static android.os.Build.TAGS;
import static android.os.Build.TYPE;
import static android.os.Build.USER;
import static android.os.Build.getRadioVersion;

public final class AuthUtils {

    public static final String UUID = getUUID();

    private static String getUUID() {
        return new UUID(getUniqueDeviceId().hashCode(), getRadioVersion().hashCode()).toString();
    }

    private static String getUniqueDeviceId() {
        return "35" +
                BOARD.length() % 10 +
                BRAND.length() % 10 +
                DEVICE.length() % 10 +
                DISPLAY.length() % 10 +
                HOST.length() % 10 +
                ID.length() % 10 +
                MANUFACTURER.length() % 10 +
                MODEL.length() % 10 +
                PRODUCT.length() % 10 +
                TAGS.length() % 10 +
                TYPE.length() % 10 +
                USER.length() % 10;
    }
}

File: app/src/main/java/com/happs/ximand/ringcontrol/model/dao/AuthUtils.kt
package com.happs.ximand.ringcontrol.model.dao

import android.os.Build
import java.util.*

object AuthUtils {

    val UUID = UUID(getUniqueDeviceId().hashCode().toLong(), Build.getRadioVersion().hashCode().toLong()).toString()

    private fun getUniqueDeviceId(): String {
        return "35" + Build.BOARD.length % 10 +
                Build.BRAND.length % 10 +
                Build.DEVICE.length % 10 +
                Build.DISPLAY.length % 10 +
                Build.HOST.length % 10 +
                Build.ID.length % 10 +
                Build.MANUFACTURER.length % 10 +
                Build.MODEL.length % 10 +
                Build.PRODUCT.length % 10 +
                Build.TAGS.length % 10 +
                Build.TYPE.length % 10 +
                Build.USER.length % 10
    }
}